variables.tf

```diff
@@ -0,0 +1,55 @@
+variable "namespace_type" {
+  type = string
+  description = "(Required) Type of namespace"
+
+  validation {
+    condition     = contains(["http", "private_dns", "public_dns"], var.namespace_type)
```

Comment:
I think is safer to go first with ONLY http. There's already a lot of testing to do aroung that. Easier to review also :).

Reply:
@lucky-edu testing results for all scenarios can be found here
https://ohpen.atlassian.net/browse/LANZ-2647
spoiler alert - it works